Dockerfile

```diff
@@ -1,14 +1,10 @@
 FROM python:3
 
-ENV UV_SYSTEM_PYTHON=true
-
 RUN apt-get update && apt-get install -y make
 
 WORKDIR /sdk
 
-COPY Makefile pyproject.toml requirements.txt requirements-dev.txt vars.mk ./
-
-RUN --mount=type=cache,mode=0755,target=/root/.cache/pip make deps
+COPY Makefile pyproject.toml vars.mk ./
```

Comment:
Do we need to include `uv.lock` here?

Reply:
I want to go with "no". It is possible that packages do not install the exact same way between local mac and docker linux.

If it is important to have the exact same setup, then we should commit uv.lock.

Reply:
It looks like it is operating system agnostic https://docs.astral.sh/uv/concepts/projects/#project-lockfile

In that case, we should be able to check it in and use it here.

Reply:
I'm still on the "no" train. If it is important to resolve to the same packages, we should use stronger package version requirements.

(I'd agree for locked-down/reproducible environments, but for open source projects its nice to have some natural chaos / updates as users will not have access to our lock file)

Reply:
Yeah, I had the same thought. I don't have strong opinions either way.